### PR TITLE
get the actual outputs of a sc call

### LINKED ISSUE #279: get the actual outputs of functions in ts caller functions

### PR DESCRIPTION
**Pull request by Elli610**
the only thing i couldn't test is the deserialization of the output because it hasn't been implemented on the SC yet
this is the part i couldn't test:
```typescript
else if(rawOutput !== null && returnType !== 'void') {
    // try to deserialize the outputs
    let output: Uint8Array = new Uint8Array(0);
    let deserializedOutput = null; 
    try{
      output = rawOutput.slice(1,-1).split(',').map(
        (s) => parseInt(s)
      ) as unknown as Uint8Array;
      deserializedOutput = eventHelper.fromBinary(output);
    }
```

closes #279

### CODE REVIEW

File: packages/massa-proto-cli/src/ts-gen.ts [hunk 1]
```diff
@@ -186,18 +187,45 @@ import {
...
 }
 
-const MASSA_EXEC_ERROR = 'massa_execution_error';
+/**
+ * This interface is used to represents the outputs of the SC call.
+ * 
+ * @see outputs - The outputs of the SC call (optional)
+ * @see events - The events emitted by the SC call (optional)
+ * @see isError - A boolean indicating wether the SC call has failed or not
+ * @see error - The error message (optional)
```
Comment (gregLibert) [inline packages/massa-proto-cli/src/ts-gen.ts:218]: Why do you plan an isError flag and an error key? Why not simply using the error key ?

File: packages/massa-proto-cli/src/ts-gen.ts [hunk 2]
```diff
@@ -214,24 +242,109 @@ let callSC: (address: string, funcName: string, binArguments: Uint8Array, maxCoi
...
+
+  // check the output and return the result
+  if (rawOutput === null && returnType !== 'void') {
+    return {
+      events: events,
+      isError: true,
+      error: 'No outputs found. Expected type: ' + returnType,
+    } as OperationOutputs;
+  }
+  else if(rawOutput === null && returnType === 'void') {
```
Comment (gregLibert) [inline packages/massa-proto-cli/src/ts-gen.ts:312]: else is useless as you have a return in the if branch

File: packages/massa-proto-cli/src/ts-gen.ts [hunk 3]
```diff
@@ -214,24 +242,109 @@ let callSC: (address: string, funcName: string, binArguments: Uint8Array, maxCoi
...
+    return {
+      events: events,
+      isError: false,
+    } as OperationOutputs;
+  }
+  else if(rawOutput !== null && returnType !== 'void') {
```
Comment (gregLibert) [inline packages/massa-proto-cli/src/ts-gen.ts:318]: same here

File: packages/massa-proto-cli/src/ts-gen.ts [hunk 4]
```diff
@@ -214,24 +242,109 @@ let callSC: (address: string, funcName: string, binArguments: Uint8Array, maxCoi
...
+        error: 'Error while deserializing the outputs: ' + err,
+      } as OperationOutputs;
+    }
+    return {
+      outputs: output,
+      events: events,
+      isError: false,
+    } as OperationOutputs;
+  }
+  else{
```
Comment (gregLibert) [inline packages/massa-proto-cli/src/ts-gen.ts:341]: same here

**Review by gregLibert — CHANGES_REQUESTED** [review]

File: packages/massa-proto-cli/src/ts-gen.ts (hunk 2 above)
Comment (Elli610) [reply packages/massa-proto-cli/src/ts-gen.ts:312]: sure

**Pull request merged**